internal/lib/logging.go

```diff
@@ -0,0 +1,47 @@
+package lib
+
+import (
+	"bytes"
+	"context"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/artifacts"
+)
+
+// LogThroughArtifactWriterIfSet reconfigures the logger used by Preflight to write to
+// the artifact writer if one is configured. This is a workaround for the library
+// implementation because existing checks make calls directly to logrus.
+func LogThroughArtifactWriterIfSet(ctx context.Context) {
+	w := artifacts.WriterFromContext(ctx)
+
+	if w != nil {
+		log.SetLevel(log.TraceLevel)
+		log.SetFormatter(&log.TextFormatter{})
+		b := bytes.NewBufferString("")
+		log.SetOutput(b)
+
+		w.WriteFile("preflight.log", b) //nolint:errcheck
+	}
+}
+
+type contextKey string
+
+var executionEnvIsCLI = contextKey("IsCLI")
+
+func CallerIsCLI(ctx context.Context) bool {
+	val := ctx.Value(executionEnvIsCLI)
+	switch b := val.(type) {
+	case bool:
+		return b
+	default:
+		return false
+	}
+}
+
+// SetCallerToCLI sets the caller as the CLI. NOTE: This is a temporary
+// workaround for internal CLI executions and will be removed at a later
+// date.
+func SetCallerToCLI(ctx context.Context) context.Context {
+	return context.WithValue(ctx, executionEnvIsCLI, true)
+}
```

Comment:
[Note To Reviewers]: This file is what lets us include some logic in the Library implementation of operator/container checks that allows us to work around our log configuration issues, where the CLI configures the logger early in the execution process.